File: src/Backdrops/AltColorgrade.cs
using Monocle;
using Celeste.Mod.Backdrops;

namespace Celeste.Mod.RainTools.Backdrops {
    [CustomBackdrop("RainTools/AltColorgrade")]
    public class AltColorgrade : Backdrop {

        public bool Enabled = true;
        public string ColorgradeA = "none";
        public string ColorgradeB = "none";
        public float LerpFactor = 0f;

        public AltColorgrade() : base() {
            Enabled = true;
            Visible = false;
        }

        public AltColorgrade(BinaryPacker.Element data) {
            Color = Calc.HexToColor(data.Attr("color")) * data.AttrFloat("alpha", 1f);

            ColorgradeA = data.Attr("colorgradeA", "none");
            ColorgradeB = data.Attr("colorgradeA", ColorgradeA);
            LerpFactor = data.AttrFloat("blendFactor");
        }

        public override void Update(Scene scene) {
            base.Update(scene);

            Enabled = Visible;
            Visible = false;
        }

    }
}

File: Loenn/effects/alt_colorgrade.lua
local alt_colorgrade = {}

alt_colorgrade.name = "RainTools/AltColorgrade"
alt_colorgrade.canForeground = true
alt_colorgrade.canBackground = false
alt_colorgrade.defaultData = {
  color = "ffffff",
  alpha = 1.0,
  colorgradeA = "none",
  colorgradeB = "none",
  blendFactor = 0.0
}
alt_colorgrade.fieldInformation = {
  color = {fieldType = "color"}
}
alt_colorgrade.fieldOrder = {
  "color", "alpha",
  "colorgradeA", "colorgradeB",
  "blendFactor"
}

return alt_colorgrade
